Userland/Libraries/LibCore/Resource.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+#include <AK/Error.h>
+#include <AK/MemoryStream.h>
+#include <AK/OwnPtr.h>
+#include <AK/Span.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Variant.h>
+#include <LibCore/File.h>
+#include <LibCore/MappedFile.h>
+
+namespace Core {
+class Resource {
+public:
+    static ErrorOr<NonnullOwnPtr<Resource>> load_from_uri(StringView);
```

Comment:
I'm not sure whether these should be refcounted or not. Probably should be? But we can always change it later when more users drive the design.